src/pal/src/numa/numa.cpp

```diff
@@ -836,8 +860,7 @@ VirtualAllocExNuma(
             if (result != NULL && g_numaAvailable)
             {
                 int nodeMaskLength = (g_highestNumaNode + 1 + sizeof(unsigned long) - 1) / sizeof(unsigned long);
-                unsigned long *nodeMask = (unsigned long*)malloc(nodeMaskLength * sizeof(unsigned long));
-
+                unsigned long *nodeMask = (unsigned long*)alloca(nodeMaskLength * sizeof(unsigned long));
```

Comment:
How large is the allocation? Can it consume the stack fully?

Reply:
It is small. That's why I have converted it to alloca. There is one bit per NUMA node in this array. Number of numa nodes is smaller or equal to the number of logical CPU cores in the system. Usually, it is much smaller. Simply said, NUMA node contains processors that have the same speed of access to a part of the memory of the system. Usually, NUMA nodes correspond to processor sockets.  E.g. RHEL 7 Linux supports max 768 logical CPU cores on amd64, so that would make this array 96 bytes large even if each processor was in its own NUMA node.

Reply:
Sounds good so far.